### DISCUSSION TITLE
Apply Schema when using layout frontmatter

### DISCUSSION THREAD

**Discussion by dlashua** [withastro/roadmap#832] [Proposal]
### Body

# Summary

Frontmatter Schema is very important when type conversions are performed. However, when the "layout" frontmatter is provided on a piece of content, it seems the schema is not applied. However, it seems that schema is intentionally not applied. (See [#9950](https://github.com/withastro/astro/issues/9950))

# Background & Motivation

Using the same layout for a `page` as well as `content` is ideal. However, if Schema is not applied then the layout must perform the conversion. But if the layout is being used in a `page` the conversion will have already been applied. The Astro Documentation even suggests [Sharing a Layout between .md and .astro files](https://docs.astro.build/en/basics/layouts/#using-one-layout-for-md-mdx-and-astro) and provides sample code to do so. However, if this feature is utilized Schema will not be applied to `content` files. 

# Goals

Ideally:
- Apply Schema Type Conversions to frontmatter when a `layout` is used in `content` files.

Alternately:
- Don't apply `layout` at all for `content` files.

**Comment by sebastienbarre** [comment]
Adding my voice here. I'm confused as to why we are providing a schema to the collection but it is not applied consistently (apparently intentionally?). 

I'm using Astro 6.1, following the tutorial. 
The `pubDate` value is in the frontmatter like this:
```
---
...
pubDate: 2022-07-15
...
---
```

If a blog post is in Markdown format, `pubDate` seems correctly coerced to a `Date` object. If my blog post is in MDX format, the value I'm getting is a `string`. Which is a problem since I want to use the same component/layout, and per the schema I should get a `Date`.

UPDATE: the issue is that I was merging my validated, type-safe frontmatter with `remarkPluginFrontmatter`. The documentation used to mention that the latter contains the raw values, but this mention [was removed](https://github.com/withastro/astro/issues/9538#issuecomment-4174227395). It is easy to shoot yourself in the foot by merging validated frontmatter with `remarkPluginFrontmatter`. This is particularly tricky to spot because it appears inconsistent between Markdown and MDX. With Markdown, my `remarkPluginFrontmatter` has `pubDate` coerced to a Date, whereas with MDX it remains a `string`, which sent me on a bit of a wild goose chase.
